### PR TITLE
feat O3-1036: Centralize workspace API and remove dependency on extension system

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This centralizes the logic involved in managing workspaces and makes them independent of the extension system.
The extension system is a way of making UI components appear, disappear, or re-order using configuration.
None of that is relevant to workspaces, so the use of the extension system was an unnecessary complication.
It also made the semantics of using the workspace very strange—we were "detaching" workspaces from the workspace window.

This also canonicalizes our vocabulary about workspaces—the forms etc that get loaded are "workspaces," the space they get loaded into is the "workspace window."

### Not included

This does not add support for [unsaved changes warnings](https://issues.openmrs.org/browse/O3-1038) nor for [launching the patient chart with a workspace open](https://issues.openmrs.org/browse/O3-1023). However those will now be very easy to implement, given the clean semantics and centralized logic of the new system.

## Reviewing

I would recommend starting with `workspaces.ts` and `workspaces.test.ts`, then `useWorkspaces.tsx`, then `workspace-window.component.tsx` and `workspace-renderer.component.tsx`, then everything else.

## Screenshots

workspaces

## Issue

https://issues.openmrs.org/browse/O3-1036

### CODE REVIEW

**Comment by brandones** [comment]
@samuelmale , please have a look at the APIs in `workspace.ts` and `useWorkspaces.ts` and see if they meet your needs (modulo the to-dos mentioned above)

File: packages/esm-patient-immunizations-app/src/immunizations/immunizations-sequence-table.tsx [hunk 1]
```diff
@@ -26,23 +26,27 @@ const SequenceTable: React.FC<SequenceTableProps> = ({ immunizations }) => {
...
-        sequences,
-        currentDose: {
-          sequenceLabel,
-          sequenceNumber,
-        },
-        vaccinationDate: existingDoses?.occurrenceDateTime,
-      });
+      launchPatientWorkspace('immunization-form-workspace');
+
+      // What's all this about?
```
Comment (brandones) [inline packages/esm-patient-immunizations-app/src/immunizations/immunizations-sequence-table.tsx:31]: @Eriq-Kibet @denniskigen @donaldkibet Do any of you know what to do with this `openWorkspaceTab` usage? It's the only one in the codebase. Is this dead code or what?

**Review by samuelmale — APPROVED** [review]
Thanks for the work well done @brandones. This looks great.

**Review by donaldkibet — APPROVED** [review]
LGTM 🚀

File: packages/esm-patient-immunizations-app/src/immunizations/immunizations-sequence-table.tsx (hunk 1 above)
Comment (brandones) [reply packages/esm-patient-immunizations-app/src/immunizations/immunizations-sequence-table.tsx:31]: Thanks @donaldkibet . Can you take care of doing that?

**Comment by denniskigen** [comment]
Really superb work, @brandones!

**Pull request merged**